compiler/rustc_passes/messages.ftl

```diff
@@ -706,7 +706,7 @@ passes_unrecognized_field =
 
 passes_unrecognized_repr_hint =
     unrecognized representation hint
-    .help = valid reprs are `C`, `align`, `packed`, `transparent`, `simd`, `i8`, `u8`, `i16`, `u16`, `i32`, `u32`, `i64`, `u64`, `i128`, `u128`, `isize`, `usize`
+    .help = valid reprs are `Rust`, `C`, `align`, `packed`, `transparent`, `simd`, `i8`, `u8`, `i16`, `u16`, `i32`, `u32`, `i64`, `u64`, `i128`, `u128`, `isize`, `usize`
```

Comment:
```suggestion
    .help = valid reprs are `Rust` (default), `C`, `align`, `packed`, `transparent`, `simd`, `i8`, `u8`, `i16`, `u16`, `i32`, `u32`, `i64`, `u64`, `i128`, `u128`, `isize`, `usize`
```
I think it's worth pointing out

Reply:
It's probably also worth it to modify that help message a bit in a separate PR, maybe even just making it multiline. It's quite verbose

Reply:
I don't know where the conversation happened, but it was explicitly wanted that every valid option is listed.  It could probably be better formatted, but the content is important.

Reply:
I've changed it to clarify that it's the default representation like was suggested. In the case that showing `repr(int)` is deemed unnecessary in the future, that can be done in a separate PR.